packages/descendants/src/useDescendant.tsx

```diff
@@ -0,0 +1,80 @@
+import {
+  ComponentProps,
+  ForwardedRef,
+  RefObject,
+  useContext,
+  useMemo,
+  useRef,
+} from 'react';
+
+import {
+  useForwardedRef,
+  useIsomorphicLayoutEffect,
+} from '@leafygreen-ui/hooks';
+
+import { DescendantContextType } from './DescendantsContext';
+import { findDescendantIndexWithId } from './utils';
+
+/** @deprecated */
+const genId = () => `_${Math.random().toString(36).substr(2, 9)}`;
+
+interface UseDescendantReturnObject<T extends HTMLElement> {
+  ref: RefObject<T>;
+  index: number;
+  id: string;
+}
+
+/**
+ * Registers a component as a descendant of a given context, and calculates its index within the rendered descendant list.
+ */
+export const useDescendant = <T extends HTMLElement>(
+  context: DescendantContextType<T>,
+  fwdRef?: RefObject<T> | ForwardedRef<T>,
+  props?: ComponentProps<any>,
+): UseDescendantReturnObject<T> => {
+  const ref: React.RefObject<T> = useForwardedRef(fwdRef ?? null, null);
```

Comment:
does this hook actually work? i feel like i've seen it not :/

Reply:
Heh, yeah I'm not sure to be honest